libraries/chain/controller.cpp

```diff
@@ -1139,7 +1139,7 @@ struct controller_impl {
 
    std::optional<block_id_type> fetch_block_id_on_head_branch_by_num(uint32_t block_num) const {
       return fork_db.apply<std::optional<block_id_type>>([&](const auto& forkdb) -> std::optional<block_id_type> {
-         auto bsp = forkdb.search_on_head_branch(block_num, include_root_t::yes);
+         auto bsp = forkdb.search_on_head_branch(block_num);
          if (bsp) return bsp->id();
```

Comment:
I think the fix should be:
https://github.com/AntelopeIO/spring/blob/12b05b1ca8a81ad42fd9351dadda539a2befee4e/libraries/chain/fork_database.cpp#L514-L516
```
      if( include_root == include_root_t::yes && root->id() == h ) {
         if (root->block_num() == block_num)
            return root;
      }

```